secp256r1/src/main/java/org/hyperledger/besu/nativelib/secp256r1/LibSECP256R1.java

```diff
@@ -113,13 +113,13 @@ public boolean verify(final byte[] dataHash, final byte[] signatureR, final byte
             return nativeSignature;
         }
 
-        // the first byte in the native representation was 0, but in the Java representation this is ignored.
+        // the first bytes in the native representation were 0, but in the Java representation this is ignored.
         // the signature needs to be shifted to the right for the native representation
-        if (signature.length == CURVE_BYTE_LENGTH - 1) {
+        if (signature.length < CURVE_BYTE_LENGTH) {
             byte[] nativeSignature = new byte[CURVE_BYTE_LENGTH];
 
             // copy signature to nativeSignature, shifting it one to the right
```

Comment:
The word `one` should be removed from the comment, as it is can be more than 1 after this change

Reply:
Done, thanks @daniellehrner